internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala

```diff
@@ -155,42 +155,59 @@ final class Dependency(val global: CallbackGlobal) extends LocateClassFile with
     private val _localInheritanceCache = HashSet.empty[ClassDependency]
     private val _topLevelImportCache = HashSet.empty[Symbol]
 
-    /** Return the enclosing class or the module class if it's a module. */
-    private def enclOrModuleClass(s: Symbol): Symbol =
-      if (s.isModule) s.moduleClass else s.enclClass
-
-    case class DependencySource(owner: Symbol) {
-      val (fromClass: Symbol, isLocal: Boolean) = {
-        val fromClass = enclOrModuleClass(owner)
-        if (fromClass == NoSymbol || fromClass.hasPackageFlag)
-          (fromClass, false)
-        else {
-          val fromNonLocalClass = localToNonLocalClass.resolveNonLocal(fromClass)
-          assert(!(fromClass == NoSymbol || fromClass.hasPackageFlag))
-          (fromNonLocalClass, fromClass != fromNonLocalClass)
-        }
+    private var _currentDependencySource: Symbol = _
+    private var _currentNonLocalClass: Symbol = _
+    private var _isLocalSource: Boolean = false
+
+    @inline def resolveNonLocalClass(from: Symbol): (Symbol, Boolean) = {
+      val fromClass = enclOrModuleClass(from)
+      if (fromClass == NoSymbol || fromClass.hasPackageFlag) (fromClass, false)
+      else {
+        val nonLocal = localToNonLocalClass.resolveNonLocal(fromClass)
+        (nonLocal, fromClass != nonLocal)
```

Comment:
Uh yeah. Hadn't realised the assert is just echoing the if condition. The original code was weird..